### PR TITLE
cc2538_rf: register with netdev

### PR DESCRIPTION
**Pull request by benpicco**
### Contribution description

In moving towards registering all 802.15.4 radios with netdev for a common init path, also convert `cc2538_rf`.
All CC2538 chips already come with an EUI-64 that was used by the driver before, so convert it to a EUI provider so that the generic code path can make use of it now.

### Testing procedure

The MAC address should stay the same as on `master`

```
2020-09-10 13:19:52,130 # Iface  6  HWaddr: 00:12  Channel: 26  Page: 0  NID: 0x23 
2020-09-10 13:19:52,134 #           Long HWaddr: 00:12:4B:00:18:F1:D6:5E 
2020-09-10 13:19:52,141 #            TX-Power: 0dBm  State: IDLE 
2020-09-10 13:19:52,144 #           AUTOACK  L2-PDU:102  MTU:1280  HL:64  RTR  
2020-09-10 13:19:52,145 #           6LO  IPHC  
2020-09-10 13:19:52,147 #           Source address length: 8
2020-09-10 13:19:52,148 #           Link type: wireless
2020-09-10 13:19:52,157 #           inet6 addr: fe80::212:4b00:18f1:d65e  scope: link  VAL
2020-09-10 13:19:52,158 #           inet6 group: ff02::2
2020-09-10 13:19:52,159 #           inet6 group: ff02::1
2020-09-10 13:19:52,160 #           inet6 group: ff02::1:fff1:d65e
```

### CODE REVIEW

**Review by fjmolinas — COMMENTED** [review]
Did some testing, seems like the result is the same but for the short address that is inverted

- master

```
2020-09-11 12:13:46,683 # ifconfig
2020-09-11 12:13:46,700 # Iface  6  HWaddr: 12:00  Channel: 26  Page: 0  NID: 0x23
2020-09-11 12:13:46,701 #           Long HWaddr: 00:12:4B:00:14:B5:B5:AF
2020-09-11 12:13:46,702 #            TX-Power: 0dBm  State: IDLE
2020-09-11 12:13:46,716 #           AUTOACK  L2-PDU:102  MTU:1280  HL:64  RTR
2020-09-11 12:13:46,716 #           6LO  IPHC
2020-09-11 12:13:46,717 #           Source address length: 8
2020-09-11 12:13:46,717 #           Link type: wireless
2020-09-11 12:13:46,732 #           inet6 addr: fe80::212:4b00:14b5:b5af  scope: link  VAL
2020-09-11 12:13:46,732 #           inet6 group: ff02::2
2020-09-11 12:13:46,733 #           inet6 group: ff02::1
2020-09-11 12:13:46,733 #           inet6 group: ff02::1:ffb5:b5af
2020-09-11 12:13:46,734 #
2020-09-11 12:13:46,734 #           Statistics for Layer 2
2020-09-11 12:13:46,747 #             RX packets 0  bytes 0
2020-09-11 12:13:46,749 #             TX packets 3 (Multicast: 3)  bytes 129
2020-09-11 12:13:46,750 #             TX succeeded 0 errors 0
2020-09-11 12:13:46,751 #           Statistics for IPv6
2020-09-11 12:13:46,752 #             RX packets 0  bytes 0
2020-09-11 12:13:46,764 #             TX packets 3 (Multicast: 3)  bytes 192
2020-09-11 12:13:46,764 #             TX succeeded 3 errors 0
2020-09-11 12:13:46,764 #
```

- pr

```
2020-09-11 12:12:33,510 # ifconfig
2020-09-11 12:12:33,511 # Iface  6  HWaddr: 00:12  Channel: 26  Page: 0  NID: 0x23
2020-09-11 12:12:33,512 #           Long HWaddr: 00:12:4B:00:14:B5:B5:AF
2020-09-11 12:12:33,526 #            TX-Power: 0dBm  State: IDLE
2020-09-11 12:12:33,527 #           AUTOACK  L2-PDU:102  MTU:1280  HL:64  RTR
2020-09-11 12:12:33,528 #           6LO  IPHC
2020-09-11 12:12:33,529 #           Source address length: 8
2020-09-11 12:12:33,530 #           Link type: wireless
2020-09-11 12:12:33,542 #           inet6 addr: fe80::212:4b00:14b5:b5af  scope: link  VAL
2020-09-11 12:12:33,543 #           inet6 group: ff02::2
2020-09-11 12:12:33,543 #           inet6 group: ff02::1
2020-09-11 12:12:33,543 #           inet6 group: ff02::1:ffb5:b5af
2020-09-11 12:12:33,544 #
2020-09-11 12:12:33,558 #           Statistics for Layer 2
2020-09-11 12:12:33,559 #             RX packets 20  bytes 860
2020-09-11 12:12:33,562 #             TX packets 15 (Multicast: 15)  bytes 645
2020-09-11 12:12:33,563 #             TX succeeded 0 errors 0
2020-09-11 12:12:33,573 #           Statistics for IPv6
2020-09-11 12:12:33,574 #             RX packets 20  bytes 1280
2020-09-11 12:12:33,575 #             TX packets 15 (Multicast: 15)  bytes 960
2020-09-11 12:12:33,576 #             TX succeeded 15 errors 0
2020-09-11 12:12:33,576 #
```

File: cpu/cc2538/radio/cc2538_rf_getset.c [hunk 1]
```diff
@@ -62,60 +62,29 @@ static const uint8_t power_lut[NUM_POWER_LEVELS] = {
...
+    addr[6] = RFCORE_FFSM_EXT_ADDR6;
+    addr[7] = RFCORE_FFSM_EXT_ADDR7;
 }
 
-uint16_t cc2538_get_addr_short(void)
+void cc2538_get_addr_short(uint8_t *addr)
 {
-    return (RFCORE_FFSM_SHORT_ADDR0 << 8) | RFCORE_FFSM_SHORT_ADDR1;
+    addr[0] = RFCORE_FFSM_SHORT_ADDR0;
+    addr[1] = RFCORE_FFSM_SHORT_ADDR1;
```
Comment (fjmolinas) [inline cpu/cc2538/radio/cc2538_rf_getset.c:80]:
```suggestion
    addr[1] = RFCORE_FFSM_SHORT_ADDR0;
    addr[0] = RFCORE_FFSM_SHORT_ADDR1;
```

to keep the same short.

**Comment by fjmolinas** [comment]
Just one comment @benpicco otherwise its all good, squash right away if you agree.

File: cpu/cc2538/radio/cc2538_rf_getset.c (hunk 1 above)
Comment (benpicco) [reply cpu/cc2538/radio/cc2538_rf_getset.c:80]: Yea then it mirrors `cc2538_set_addr_short()`

But I just noticed that the same issue must be present with `cc2538_get_addr_long()` / `cc2538_set_addr_long()` as there I do

    addr[0] = RFCORE_FFSM_EXT_ADDR0;

and

    RFCORE_FFSM_EXT_ADDR0 = addr[7];

**Comment by benpicco** [comment]
Fixed, verified that `txtsnd` also works with both long and short addresses.

**Review by fjmolinas — APPROVED** [review]
ACK, please squash @benpicco

**Pull request merged**